perf_test/sparse/KokkosSparse_sptrsv_aux.hpp

```diff
@@ -0,0 +1,626 @@
+  using scalar_t = typename values_view_t::value_type;
+
+  using STS = Kokkos::Details::ArithTraits<scalar_t>;
+  using range_type = Kokkos::pair<int, int>;
+
+  auto graph = A.graph; // in_graph
+  int n = graph.numRows ();
+  auto row_map = graph.row_map;
+  auto entries = graph.entries;
+  auto values  = A.values;
+
+  row_map_view_host_t hr = Kokkos::create_mirror_view (row_map);
+  cols_view_host_t    hc = Kokkos::create_mirror_view (entries);
+  values_view_host_t  hv  = Kokkos::create_mirror_view (values);
+
+  Kokkos::deep_copy (hr, row_map);
+  Kokkos::deep_copy (hc, entries);
+  Kokkos::deep_copy (hv, values);
+
+  // compress
+  int nnzA0 = hr (n);
+  int nnzA = 0;
+  for (int i = 0; i < n; i++) {
+    int nnz = nnzA;
+    for (int k = hr (i); k < hr (i+1); k++) {
+      if (hv(k) != STS::zero ()) {
+        hv (nnzA) = hv (k);
+        hc (nnzA) = hc (k); 
+        nnzA++;
+      }
+    }
+    hr (i) = nnz;
+  }
+  hr (n) = nnzA;
+  std::cout << "   > compressed from " << nnzA0 << " to " << nnzA << " nnzs" << std::endl;
+
+  // allocate & create
+  row_map_view_t new_row_map ("rowmap_view", n+1);
+  cols_view_t    new_entries ("colmap_view", nnzA);
+  values_view_t  new_values  ("values_view", nnzA);
+
+  Kokkos::deep_copy (new_row_map, hr);
+  Kokkos::deep_copy (new_entries, subview (hc, range_type(0, nnzA)));
+  Kokkos::deep_copy (new_values,  subview (hv, range_type(0, nnzA)));
+
+  graph_t static_graph (new_entries, new_row_map);
+  crsmat_t crsmat ("CrsMatrix", n, new_values, static_graph);
+  return crsmat;
+}
+
+
+/* ========================================================================================= */
+template <typename crsmat_t, typename host_crsmat_t>
+bool check_cusparse(host_crsmat_t &Mtx, bool col_majorL, crsmat_t &L, bool col_majorU, crsmat_t &U,
+                    int *perm_r, int *perm_c, double tol, int loop) {
+#if defined(KOKKOSKERNELS_ENABLE_TPL_CUSPARSE) 
+  using values_view_t = typename crsmat_t::values_type::non_const_type;
+  using scalar_t      = typename values_view_t::value_type;
+
+  using host_values_view_t = typename host_crsmat_t::values_type::non_const_type;
+  using host_scalar_t      = typename host_values_view_t::value_type;
+
+  using execution_space = typename values_view_t::execution_space;
+  using memory_space    = typename execution_space::memory_space;
+
+  using host_execution_space = typename host_values_view_t::execution_space;
+  using host_memory_space    = typename host_execution_space::memory_space;
+
+  using host_scalar_view_t = Kokkos::View< scalar_t*, host_memory_space >;
+  using      scalar_view_t = Kokkos::View< scalar_t*,      memory_space >;
+
+  scalar_t ZERO = scalar_t (0);
+  scalar_t ONE = scalar_t (1);
+
+  Kokkos::Timer timer;
+  const int nrows = Mtx.graph.numRows ();
+
+  // ==============================================
+  // > create a handle
+  cusparseStatus_t status;
+  cusparseHandle_t handle = 0; 
+  status = cusparseCreate (&handle);
```

Comment:
Doesn't kokkos-kernels already have a wrapper for cuSPARSE handles?  It would make sense to work with that existing framework.